flang/lib/Optimizer/Analysis/AliasAnalysis.cpp

```diff
@@ -372,6 +372,36 @@ getAttrsFromVariable(fir::FortranVariableOpInterface var) {
   return attrs;
 }
 
+static std::optional<omp::BlockArgOpenMPOpInterface>
+getOpenMPBlockArgInterface(Operation *op) {
+  if (!op)
+    return {};
+  return dyn_cast<omp::BlockArgOpenMPOpInterface>(op);
```

Comment:
nit: this can be simpler (maybe we don't need the helper now?)
```suggestion
  return dyn_cast_if_present<omp::BlockArgOpenMPOpInterface>(op);
```
https://llvm.org/docs/ProgrammersManual.html#the-isa-cast-and-dyn-cast-templates